### PR TITLE
HaikuWebSearch: Initial recipe

### PR DESCRIPTION
**Pull request by tts2k**

### CODE REVIEW

File: haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:gcc
+	cmd:make
+	"
+
+USER_SETTING_FILES="
+	settings/HaikuWebSearch directory
+	settings/HaikuWebSearch/engines
+	"
```
Comment (alaviss) [inline haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe]: HaikuWebSearch only [looks](https://github.com/Oyami-Srk/HaikuWebSearch/blob/master/SearchView.cpp#L96) for engines in `B_USER_SETTINGS_DIRECTORY`. This means `USER_SETTING_FILES` (actually it's `USER_SETTINGS_FILES`) is not needed as this only applies to files within `B_SYSTEM_SETTINGS_DIRECTORY`

File: haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe [hunk 2]
```diff
@@ -0,0 +1,48 @@
+SUMMARY="A search application"
+DESCRIPTION="A Haiku Application that provides a replicant to search the internet."
+HOMEPAGE="https://github.com/Oyami-Srk/HaikuWebSearch"
+COPYRIGHT="2017 Shiroko"
+LICENSE="MIT"
+REVISION="1"
+commit="c9bc31b35f066870cc73bca78c3f02ac331b1d3f"
```
Comment (alaviss) [inline haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe]: We typically use `srcGitRev`

```diff
-commit="c9bc31b35f066870cc73bca78c3f02ac331b1d3f"
+srcGitRev="c9bc31b35f066870cc73bca78c3f02ac331b1d3f"

File: haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+SOURCE_URI="$HOMEPAGE/archive/$commit.tar.gz"
+CHECKSUM_SHA256="86ccf3e816fac123ac156bb54678001506ddb59e292c3b2a3f4ea36fd9719de1"
+SOURCE_FILENAME="HaikuWebSearch-$portVersion-$commit.tar.gz"
+SOURCE_DIR="HaikuWebSearch-$commit"
```
Comment (alaviss) [inline haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe]:
```diff
-SOURCE_URI="$HOMEPAGE/archive/$commit.tar.gz"
+SOURCE_URI="$HOMEPAGE/archive/$srcGitRev.tar.gz"
 CHECKSUM_SHA256="86ccf3e816fac123ac156bb54678001506ddb59e292c3b2a3f4ea36fd9719de1"
-SOURCE_FILENAME="HaikuWebSearch-$portVersion-$commit.tar.gz"
-SOURCE_DIR="HaikuWebSearch-$commit"
+SOURCE_FILENAME="HaikuWebSearch-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="HaikuWebSearch-$srcGitRev"

File: haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+
+PROVIDES="
+	HaikuWebSearch = $portVersion
```
Comment (alaviss) [inline haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe]:
```diff
-	HaikuWebSearch = $portVersion
+	haikuwebsearch = $portVersion

File: haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+DESCRIPTION="A Haiku Application that provides a replicant to search the internet."
+HOMEPAGE="https://github.com/Oyami-Srk/HaikuWebSearch"
+COPYRIGHT="2017 Shiroko"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="c9bc31b35f066870cc73bca78c3f02ac331b1d3f"
+SOURCE_URI="$HOMEPAGE/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="86ccf3e816fac123ac156bb54678001506ddb59e292c3b2a3f4ea36fd9719de1"
+SOURCE_FILENAME="HaikuWebSearch-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="HaikuWebSearch-$scrGitRev"
```
Comment (fbrosson) [inline haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe]:
```diff
-SOURCE_DIR="HaikuWebSearch-$scrGitRev"
+SOURCE_DIR="HaikuWebSearch-$srcGitRev"
```
(scr -> src)

BTW, I would recommend a rename of the recipe to `haikuwebsearch-0~git.recipe`.

File: haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	"
+
+USER_SETTING_FILES="
+	settings/HaikuWebSearch directory
+	settings/HaikuWebSearch/engines
+	"
+
+BUILD()
+{
+	make
```
Comment (fbrosson) [inline haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe]:
```diff
-	make
+	make OBJ_DIR=objects
```

File: haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir
+	cp objects.*-release/Haiku-WebSearch $appsDir/
```
Comment (fbrosson) [inline haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe]:
```diff
 INSTALL()
 {
-	mkdir -p $appsDir
+	install -d $appsDir
-	cp objects.*-release/Haiku-WebSearch $appsDir/
+	install -t $appsDir objects/Haiku-WebSearch
-	addAppDeskbarSymlink $appsDir/Haiku-WebSearch "HaikuWebSearch"
+	addAppDeskbarSymlink $appsDir/Haiku-WebSearch HaikuWebSearch
 }
```

File: haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe (hunk 5 above)
Comment (tts2k) [reply haiku-apps/haikuwebsearch/haikuwebsearch-0.0.0~git.recipe]: I noticed this a few hours ago but forgot to fix lol

**Comment by fbrosson** [comment]
@tts2k, regarding the LIBS variable in the project's Makefile, I can confirm with this change in [Line 58](https://github.com/Oyami-Srk/HaikuWebSearch/blob/master/Makefile#L58):
```diff
-LIBS = /boot/system/lib/libroot.so /boot/system/lib/libbe.so /boot/system/lib/libsupc++.so /boot/system/lib/libtranslation.so /boot/system/lib/libstdc++.so
+LIBS = be root $(STDCPPLIBS)
```
the app builds fine seems to work fine too, on both x86_gcc2 and x86_64.

@Oyami-Srk might accept to do this change for you, directly, as he is the author of the app.
Alternatively, you can open a PR on https://github.com/Oyami-Srk/HaikuWebSearch

File: haiku-apps/haikuwebsearch/haikuwebsearch-0~git.recipe [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+COPYRIGHT="2017 Shiroko"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="c9bc31b35f066870cc73bca78c3f02ac331b1d3f"
+SOURCE_URI="$HOMEPAGE/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="86ccf3e816fac123ac156bb54678001506ddb59e292c3b2a3f4ea36fd9719de1"
+SOURCE_FILENAME="HaikuWebSearch-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="HaikuWebSearch-$srcGitRev"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
```
Comment (fbrosson) [inline haiku-apps/haikuwebsearch/haikuwebsearch-0~git.recipe]:
```diff
-ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+ARCHITECTURES="x86_gcc2 ?x86 x86_64"
```
(The app also builds and runs on x86_gcc2 so we can mark it as tested :)

**Pull request merged**